tests/org.palladiosimulator.dataflow.confidentiality.analysis.tests/src/org/palladiosimulator/dataflow/confidentiality/analysis/ConstraintTest.java

```diff
@@ -61,12 +70,13 @@ private Stream<Arguments> violationTestProvider() {
      * @return Returns true, if the constraint is violated. Otherwise, the method returns false.
      */
     private boolean travelPlannerCondition(AbstractActionSequenceElement<?> node) {
-        List<Literal> assignedRoles = node.getNodeCharacteristicsWithName("AssignedRoles");
-        List<Literal> grantedRoles = node.getDataFlowCharacteristicsWithName("GrantedRoles");
+    	List<Literal> assignedRoles = node.getNodeCharacteristicsWithName("AssignedRoles");
+    	Map<DataFlowVariable, List<Literal>> grantedRoles = node.getDataFlowCharacteristicsWithName("GrantedRoles");
+    	
         printNodeInformation(node);
-
-        return assignedRoles.stream()
-            .noneMatch(ar -> grantedRoles.contains(ar));
+        
+        return grantedRoles.isEmpty() || grantedRoles.values().stream()
+        .allMatch(assignedRoles::containsAll);
```

Comment:
Is this correct? Access control is also satisfied if at least one role is in the set intersections, its not required that all roles match

Reply:
The results look different. Current results:
```
Constraint violation found: CallingUserActionSequenceElement / calling (store ccd, _gdSegPVhEeuMKba1Qn68bg))
	Node characteristics: AssignedRoles.User
	Data flow Variables:  ccd [GrantedRoles.User]

Constraint violation found: DatabaseActionSequenceElement / writing (CreditCardDataDB, _60Fv8PVdEeuMKba1Qn68bg))
	Node characteristics: AssignedRoles.User
	Data flow Variables:  ccd [GrantedRoles.User]

Constraint violation found: CallingUserActionSequenceElement / returning (store ccd, _gdSegPVhEeuMKba1Qn68bg))
	Node characteristics: AssignedRoles.User
	Data flow Variables:  
```

Expected results (after I changed the constraint the last time):
```
Constraint violation found: SEFFActionSequenceElement (Begining book, _vorK8fVeEeuMKba1Qn68bg))
	Node characteristics: AssignedRoles.Airline
	Data flow Variables:  flight [GrantedRoles.User, GrantedRoles.Airline], ccd [GrantedRoles.User]

Constraint violation found: SEFFActionSequenceElement (return confirmation, _7HCu4PViEeuMKba1Qn68bg))
	Node characteristics: AssignedRoles.Airline
	Data flow Variables:  flight [GrantedRoles.User, GrantedRoles.Airline], ccd [GrantedRoles.User], RETURN [GrantedRoles.User, GrantedRoles.Airline]
```

Reply:
After wondering about my constraint, i've fixed my bad refactoring here. I plan on implementing some further tests in #26 to catch wrong results